src/osx/carbon/frame.cpp

```diff
@@ -300,7 +300,10 @@ void wxFrame::SetToolBar(wxToolBar *toolbar)
 #ifndef __WXOSX_IPHONE__
 #if wxOSX_USE_NATIVE_TOOLBAR
     if ( toolbar )
+    {
         toolbar->MacInstallNativeToolbar( true ) ;
+        toolbar->Realize();
```

Comment:
right now the call to Realize would be executed twice, in the 'ordinary' sequence, this could be guarded with something like `bool needsRealize = !m_frameToolBar;` which could be set in front of the `m_frameToolBar = toolbar`

Reply:
I'd rather call `Realize()` on demand in `MacInstallNativeToolbar()` then.

Reply:
@vadz @csomor ,
I will see what can be done about that...

Thx.